### PR TITLE
Initial data modelling work

### PR DESCRIPTION
**Pull request by fredex42**
Creation of an Event data model, stored in dynamo DB, with create and view functionality via form submit.  Also adding some graphics and initial styling

### CODE REVIEW

File: app/controllers/Events.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      "start_time"->jodaDate,
+      "channel_id"->nonEmptyText,
+      "creator"->nonEmptyText,
+      "contacts"->seq(text),
+      "output_channels"->seq(text),
+      "status"->text
+    )(Event.apply)(Event.unapply)
+  )
+
+  def add_form = Action {
```
Comment (akash1810) [inline app/controllers/Events.scala:25]: minor: camelCase is preferred with scala

File: app/views/eventlist.scala.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
+@(events: List[Event])
+@base(title = "Live Events") {
+
+    <table class="dashboardpanel">
```
Comment (akash1810) [inline app/views/eventlist.scala.html:4]: why table? could we use flexbox instead?

File: cfn/cfn.json [hunk 3]
```diff
@@ -12,6 +12,14 @@
         "InstanceKey": {
             "Type": "String",
             "Description": "Name of a pre-created keypair to associate with any instances created"
+        },
+        "Region": {
+            "Type": "String",
+            "Description": "Amazon region where we are launching"
+        },
+        "AccountID": {
```
Comment (akash1810) [inline cfn/cfn.json]: could we use the [pseudo parameter](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html) instead?

File: cfn/cfn.json [hunk 4]
```diff
@@ -12,6 +12,14 @@
         "InstanceKey": {
             "Type": "String",
             "Description": "Name of a pre-created keypair to associate with any instances created"
+        },
+        "Region": {
```
Comment (akash1810) [inline cfn/cfn.json]: could we use the [pseudo parameter](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html) here?

File: conf/livestreamcreator.properties [hunk 5]
```diff
@@ -0,0 +1 @@
+com.gu.livestreamcreator.events_dynamo_table = "live-stream-creator-EventModelTable-VVXLQ8GMND5V"
```
Comment (akash1810) [inline conf/livestreamcreator.properties]: Don't think this needs to be checked in?

File: project/plugins.sbt [hunk 6]
```diff
@@ -2,7 +2,9 @@ logLevel := Level.Info
 
 resolvers += "Typesafe repository" at "http://repo.typesafe.com/typesafe/releases/"
 
-addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.3.8")
+libraryDependencies += "com.typesafe.play" %% "anorm" % "2.4.0"
```
Comment (akash1810) [inline project/plugins.sbt]: why is this here rather than `build.sbt`?

File: app/assets/dashboard.css [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+    display:block;
+    width:100%;
+    height:auto;
+}
+
+.selector_menu {
+    border: none;
+    border-top: none;
+    border-left: none;
+    border-right: none;
```
Comment (akash1810) [inline app/assets/dashboard.css]: don't think you need all the above rules

File: app/assets/dashboard.css [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+    border-left: 3px solid #0000aa;
+    text-decoration:underline;
+}
+
+.selectable:hover {
+   background-color: #CCCCCC;
+   cursor:pointer;
+}
+
+.status_success {
```
Comment (akash1810) [inline app/assets/dashboard.css]: `status--success` would be better [BEM](http://csswizardry.com/2013/01/mindbemding-getting-your-head-round-bem-syntax/) (similar for classes below too)

File: app/assets/master.css [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+ }
+
+div.login_bar {
+ z-index: 99999;
+ float:right; font-size:0.9em; border:1px;
+ line-height: 50px;	//size of the google+ user image
+ vertical-align:top;
+}
+
+.error { color:#FF0000; }
```
Comment (akash1810) [inline app/assets/master.css:5]: We may want to add a linting step to the build for css as the different styles are being used - would be great to standardise on a single, common one. For example, selectors and rules should be consistently new-lined.

File: app/views/eventlist.scala.html (hunk 2 above)
Comment (fredex42) [reply app/views/eventlist.scala.html:4]: absolutely... I've not used a flexbox before though. can we accept this then PR an update next?

File: cfn/cfn.json (hunk 3 above)
Comment (fredex42) [reply cfn/cfn.json]: will do

File: conf/livestreamcreator.properties (hunk 5 above)
Comment (fredex42) [reply conf/livestreamcreator.properties]: no this was checked in by accident, it's not used now.

File: project/plugins.sbt (hunk 6 above)
Comment (fredex42) [reply project/plugins.sbt]: i'll try it in build.sbt.  i was mucking around for ages after upping to 2.5 and it worked there

File: app/assets/dashboard.css (hunk 7 above)
Comment (fredex42) [reply app/assets/dashboard.css]: no, they were copied over from another project

File: app/assets/master.css (hunk 9 above)
Comment (fredex42) [reply app/assets/master.css:5]: yes that makes sense. would it be a good idea to do that as a seperate PR and pull it in at cloudformation, or to do it in a pre-commit hook?

File: app/controllers/Events.scala (hunk 1 above)
Comment (fredex42) [reply app/controllers/Events.scala:25]: can we leave this for the time being and replace later?

File: cfn/cfn.json [hunk 10]
```diff
@@ -321,6 +321,32 @@
                                         "s3:GetObjectAcl"
                                     ],
                                     "Resource": "arn:aws:s3:::gnm-multimedia-rr-deployables/*"
+                                },
```
Comment (akash1810) [inline cfn/cfn.json:324]: cfn has been moved to the private repo now as it has (private) details on the infrastructure. You'll need to rebase master, remove these changes and issue a PR in the [infra repo](https://github.com/guardian/live-stream-creator-infra).

File: app/assets/master.css (hunk 9 above)
Comment (akash1810) [reply app/assets/master.css:5]: > would it be a good idea to do that as a seperate PR

would prefer to do it early as css has a tendency to be left alone once its been written for fear of breaking things in super subtle ways... Your call.

I'd do the linting as part of the CI process. Personally I find pre-commit hooks annoying, for example when you're just hacking away and want to push a WIP branch up... There's also no guarantee they'll be added to everyone's version of the repo too. Again, your call.
